firefox-ios/Client/Frontend/Home/LogoHeader/HomepageHeaderCell.swift

```diff
@@ -0,0 +1,121 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import UIKit
+import Common
+
+struct HomepageHeaderCellViewModel {
+    var hidePrivateModeButton: Bool
+    var isPrivate: Bool
+
+    private var action: (() -> Void)
+    private var homepageTelemetry = HomepageTelemetry()
+
+    init(isPrivate: Bool, hidePrivateModeButton: Bool, action: @escaping () -> Void) {
+        self.isPrivate = isPrivate
+        self.hidePrivateModeButton = hidePrivateModeButton
+        self.action = action
+    }
+
+    func switchMode() {
+        action()
+        homepageTelemetry.sendHomepageTappedTelemetry(enteringPrivateMode: !isPrivate)
+    }
+}
+
+// Header for the homepage in both normal and private mode
+// Contains the firefox logo and the private browsing shortcut button
+class HomepageHeaderCell: UICollectionViewCell, ReusableCell {
+    enum UX {
+        static let iPhoneTopConstant: CGFloat = 16
+        static let iPadTopConstant: CGFloat = 54
+        static let circleSize = CGRect(width: 40, height: 40)
+    }
+
+    var viewModel: HomepageHeaderCellViewModel?
+    var actionButtonAction: (() -> Void)?
+
+    private lazy var stackContainer: UIStackView = .build { stackView in
+        stackView.axis = .horizontal
+        stackView.distribution = .fill
+    }
+
+    private lazy var logoHeaderCell: HomeLogoHeaderCell = {
+        let logoHeader = HomeLogoHeaderCell()
+        return logoHeader
+    }()
+
+    private lazy var circularView: UIView = .build { view in
+        view.frame = UX.circleSize
+        view.backgroundColor = UIColor.white
+        view.layer.cornerRadius = view.frame.size.width / 2
+        view.isUserInteractionEnabled = true
+        view.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.switchMode)))
+    }
+
+    private lazy var privateModeButton: UIButton = .build { button in
+        let maskImage = UIImage(named: ImageIdentifiers.privateMaskSmall)?.withRenderingMode(.alwaysTemplate)
+        button.setImage(maskImage, for: .normal)
+        button.imageView?.contentMode = .scaleAspectFit
+        button.accessibilityLabel = .TabTrayToggleAccessibilityLabel
+        button.accessibilityHint = .TabTrayToggleAccessibilityHint
+    }
+
+    // MARK: - Initializers
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupView()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - UI Setup
+
+    func setupView() {
+        let isiPad = UIDevice.current.userInterfaceIdiom == .pad
+        let topAnchorConstant = isiPad ? UX.iPadTopConstant : UX.iPhoneTopConstant
+        privateModeButton.insertSubview(circularView, belowSubview: privateModeButton.imageView ?? privateModeButton)
```

Comment:
I am very iffy about this piece of code, let me know if you see a better suggestion in placing the circle view behind the image view. Leaving this PR in draft for now as I may come with fresher eyes tomorrow, but would like to get some early feedback in general!

Reply:
refactored and removed this code